tests/subsys/net/lib/lwm2m/lwm2m_rd_client/CMakeLists.txt

```diff
@@ -0,0 +1,33 @@
+cmake_minimum_required(VERSION 3.20.0)
+
+find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
+project(lwm2m_rd_client)
+
+set(APP_SRC_DIR ${CMAKE_CURRENT_SOURCE_DIR}/src)
+
+# Add test sources
+target_sources(app PRIVATE ${APP_SRC_DIR}/main.c)
+target_sources(app PRIVATE ${APP_SRC_DIR}/stubs.c)
+target_sources(app PRIVATE ${ZEPHYR_BASE}/../zephyr/subsys/net/lib/lwm2m/lwm2m_rd_client.c)
+
+# Add includes directories
+target_include_directories(app PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/src)
+target_include_directories(app PRIVATE ${ZEPHYR_BASE}/../nrf/include/)
+target_include_directories(app PRIVATE ${ZEPHYR_BASE}/include/)
+target_include_directories(app PRIVATE ${ZEPHYR_BASE}/../nrfxlib/nrf_modem/include/)
+target_include_directories(app PRIVATE ${ZEPHYR_BASE}/subsys/net/lib/lwm2m/)
+
+add_compile_definitions(CONFIG_LWM2M_RD_CLIENT_AUTO_INIT=n)
```

Comment:
This looks wrong, you shouldn't declare boolean Kconfigs like this, if you want to mimic `CONFIG_LWM2M_RD_CLIENT_AUTO_INIT=n` just don't define it.